app/src/main/java/com/simplecity/amp_library/ui/presenters/PlayerPresenter.java

```diff
@@ -154,8 +163,13 @@ private void updateFavorite(boolean isFavorite) {
     public void updateTrackInfo() {
         PlayerView view = getView();
         if (view != null) {
+            totalTime = MusicUtils.getDuration();
             view.trackInfoChanged(MusicUtils.getSong());
             view.currentTimeChanged(MusicUtils.getPosition() / 1000);
+            if(SettingsManager.getInstance().preferRemainingTime())             //can be refactored into another function
+                view.totalTimeChanged(MusicUtils.getDuration()/1000 - MusicUtils.getPosition()/1000);
```

Comment:
1. Why are we using `MusicUtils.getDuration()` here, when we have already stored it in the `totalTime` variable? *This comment can be ignored because the `totalTime` variable will be removed as per other comments,

2. `MusicUtils.getDuration()/1000 - MusicUtils.getPosition()/1000` can be simplified to `(MusicUtils.getDuration()- MusicUtils.getPosition()) / 1000`

3. Formatting: Use spaces around mathematical operators (`x / y` instead of `x/y`)